pretty/src/geometry/rect.rs

```diff
@@ -5,14 +5,41 @@ use super::range::Range;
 
 /// A rectangle, either on the screen, or on the document.
 /// Includes its upper-left, but excludes its lower-right.
-/// (Used in this file only, to simplify the implementation of Regions.)
-#[derive(PartialEq, Eq, Clone, Copy)]
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
 pub struct Rect {
     pub rows: Range<Row>,
     pub cols: Range<Col>,
 }
 
+/// Iterator over sub-rectangles of a Rect that was split horizontally
+pub struct SplitRectHorzIter<'a> {
+    remaining: Rect,
+    widths: &'a [Col],
+}
+
+/// Iterator over sub-rectangles of a Rect that was split vertically
+pub struct SplitRectVertIter<'a> {
+    remaining: Rect,
+    heights: &'a [Row],
+}
+
 impl Rect {
+    /// Create a new Rect with the given position and size
+    pub fn new(top_left: Pos, size: Pos) -> Rect {
```

Comment:
TODO: find places that constructed `Rect`s manually, and see if they'd be better expressed using this method

Reply:
done